GDJS/Runtime/runtimeobject.ts

```diff
@@ -1617,40 +1618,128 @@ namespace gdjs {
+        RuntimeObject.separateFromObjectsStatics.moveXArray;
+      let moveYArray: Array<float> =
+        RuntimeObject.separateFromObjectsStatics.moveYArray;
+      moveXArray.length = 0;
+      moveYArray.length = 0;
+
       const hitBoxes = this.getHitBoxes();
+
       for (const name in objectsLists.items) {
         if (objectsLists.items.hasOwnProperty(name)) {
-          const objects = objectsLists.items[name];
-
-          //Check if their is a collision with each object
-          for (let i = 0, len = objects.length; i < len; ++i) {
-            if (objects[i].id != this.id) {
-              const otherHitBoxes = objects[i].getHitBoxes();
-              for (let k = 0, lenk = hitBoxes.length; k < lenk; ++k) {
-                for (let l = 0, lenl = otherHitBoxes.length; l < lenl; ++l) {
-                  const result = gdjs.Polygon.collisionTest(
-                    hitBoxes[k],
-                    otherHitBoxes[l],
-                    ignoreTouchingEdges
-                  );
-                  if (result.collision) {
-                    xMove += result.move_axis[0];
-                    yMove += result.move_axis[1];
-                    moved = true;
-                  }
+          const otherObjects = objectsLists.items[name];
+
+          // Check if their is a collision with each object
+          for (const otherObject of otherObjects) {
+            if (otherObject.id === this.id) {
+              continue;
+            }
+            const otherHitBoxes = otherObject.getHitBoxes();
+            for (const hitBox of hitBoxes) {
+              for (const otherHitBox of otherHitBoxes) {
+                const result = gdjs.Polygon.collisionTest(
+                  hitBox,
+                  otherHitBox,
+                  ignoreTouchingEdges
+                );
+                if (result.collision) {
+                  moveXArray.push(result.move_axis[0]);
+                  moveYArray.push(result.move_axis[1]);
                 }
               }
             }
           }
         }
       }
+      return this._doSeparateFromObjects(moveXArray, moveYArray);
+    }
+
+    /**
+     * Separate the object from others objects, using the results from collisionTest call.
+     * @param moveXArray
+     * @param moveYArray
+     * @return true if the object was moved
+     */
+    private _doSeparateFromObjects(
+      moveXArray: Array<float>,
+      moveYArray: Array<float>
+    ): boolean {
+      if (moveXArray.length === 0) {
+        moveXArray.length = 0;
+        moveYArray.length = 0;
+        return false;
+      }
+      if (moveXArray.length === 1) {
+        // Move according to the results returned by the collision algorithm.
+        this.setPosition(
+          this.getX() + moveXArray[0],
+          this.getY() + moveYArray[0]
+        );
+        moveXArray.length = 0;
+        moveYArray.length = 0;
+        return true;
+      }
+
+      // Find the longest vector
+      let distanceSqMax = 0;
```

Comment:
I don't know if I am the only one who has a problem with abbreviation, but I think it would be clearer with `distanceSq` => `squaredDistance` and `distanceSqMax` => `maxSquaredDistance`

Reply:
Yes, I don't like them either. I only use a few that think are common like "Sq" or "Itr". I will make the change but I will keep the max at the end because it's a maximum of the distances and not a distance of a maximum.

Reply:
I guess abbreviations are always obvious to the one using them (and myself from the past is not the same as myself in the future!)
Ok for the max! Thanks